### PR TITLE
PS-7895 arch_oper_mutex is not released inside Arch_Page_Sys::track_page

### PR DESCRIPTION
**Pull request by rahulmalik87**
https://jira.percona.com/browse/PS-7895

Problem:

Hang in checkpoint flusher thread due to arch_oper_mutex can't be acquired.
It can happen when page-tracking in enabled and server is under heavy load.

The Arch_Page_Sys::track_page is supposed to be released and reacquire the
archive mutex through function wait_flush_archiver. If is_flushable is false
it never enters in while loop in wait_flush_archiver.
so arch_oper_mutex is never relased.

Fix:

Release and reacquire arch_oper_mutex inside Arch_Page_Sys::track_page

### CODE REVIEW

**Review by satya-bodapati — APPROVED** [review]

File: storage/innobase/arch/arch0page.cc
Comment (venkatesh-prasad-v) [inline storage/innobase/arch/arch0page.cc]: Not an issue.

FYI, `sentry` objects can be used in such cases to ensure that the lock is released when the control gets out of scope.

Usage is as below
```
#include "raii/sentry.h" /* raii::Sentry */
...
...
arch_oper_mutex_enter();
/* Create a sentry object to call `arch_oper_mutex_exit()` on exit */
raii::Sentry<> arch_oper_mutex_guard{[&]() { arch_oper_mutex_exit(); }};
```

Comment (satya-bodapati) [reply storage/innobase/arch/arch0page.cc]: Good suggestion. no need to create dummy class for RAII

**Review by venkatesh-prasad-v — APPROVED** [review]
Looks good. Please wait for the final approval from @satya-bodapati to merge it.

**Review by satya-bodapati — COMMENTED** [review]
@rahulmalik87 did you test your fix with the latest changes (sentry object). Please check if the hang is not reproducible.

**Comment by rahulmalik87** [comment]
Could not reproduce the bug on latest 8.0.27 branch

**Pull request closed without merging**
